components/sup/src/manager/service/config.rs

```diff
@@ -81,12 +82,17 @@ impl ServiceConfig {
     pub fn new(package: &PackageInstall,
                mgr_cfg: &ManagerConfig,
                runtime_cfg: &RuntimeConfig,
-               config_root: Option<PathBuf>,
+               config_root_from_config: Option<PathBuf>,
                bindings: Vec<(String, ServiceGroup)>)
                -> Result<ServiceConfig> {
-        let config_root = config_root.unwrap_or(package.installed_path.clone());
+        let config_root = match config_root_from_config {
```

Comment:
These 4 lines do the same as the replaced line except they also cost an additional allocation. Why the change?